src/digid_eherkenning_oidc_generics/views.py

```diff
@@ -1,24 +1,151 @@
+
+    Specify this as a kwarg in the ``as_view(config_class=...)`` class method.
+    """
+
+    def get_settings(self, attr, *args):  # type: ignore
+        if (config := getattr(self, "_config", None)) is None:
+            # django-solo and type checking is challenging, but a new release is on the
+            # way and should fix that :fingers_crossed:
+            config = cast(T, self.config_class.get_solo())
+            self._config = config
+        return get_setting_from_config(config, attr, *args)
+
+    def get(
+        self, request: HttpRequest, return_url: str = "", *args, **kwargs
+    ) -> HttpResponseRedirect:
+        if not return_url:
+            raise ValueError("You must pass a return URL")
+
+        url_is_safe = url_has_allowed_host_and_scheme(
+            url=return_url,
+            allowed_hosts=request.get_host(),
+            require_https=request.is_secure(),
+        )
+        if not url_is_safe or not return_url:
+            raise DisallowedRedirect(f"Can't redirect to '{return_url}'")
+
+        self._check_idp_availability()
+
+        # We add our own key to keep track of the redirect URL. In the case of
+        # authentication failure (or canceled logins), the session is cleared by the
+        # upstream library, so in the callback view we store this URL so that we know
+        # where to redirect with the error information.
+        self.request.session["of_redirect_next"] = return_url
+
+        response = super().get(request, *args, **kwargs)
+
+        # mozilla-django-oidc grabs this from request.GET and since that is not mutable,
+        # it's easiest to just override the session key with the correct value.
+        request.session["oidc_login_next"] = return_url
+
+        return response
+
+    def _check_idp_availability(self) -> None:
+        endpoint = self.OIDC_OP_AUTH_ENDPOINT
+        try:
+            # Verify that the identity provider endpoint can be reached. This is where
+            # the user ultimately gets redirected to.
+            response = requests.get(endpoint)
+            if response.status_code > 400:
+                response.raise_for_status()
+        except requests.RequestException as exc:
+            logger.info(
+                "OIDC provider endpoint '%s' could not be retrieved",
+                endpoint,
+                exc_info=exc,
+            )
+            raise OIDCProviderOutage() from exc
+
+    def get_extra_params(self, request: HttpRequest) -> dict[str, str]:
+        """
+        Add a keycloak identity provider hint if configured.
+        """
+        extra = super().get_extra_params(request)
+        options = self.config_class._meta
+        # Store which config class to use in the params so that the callback view can
+        # extract this again.
+        # TODO: verify this cannot be tampered with!
+        extra["config"] = f"{options.app_label}.{options.object_name}"
```

Comment:
Correct! This will be consumed by the (single) callback view to handle the different configs. I'm working on that right now and I have a theory I need to confirm about whether it can be spoofed or not. If my theory is wrong, we can move this into the session instead.

I think if the user tampers with this, it will break anyway because it will point to a different client ID + secret to exchange the `code` param for a userinfo token, which should fail unless the same credentials are used in different configs (but the latter seems like a problem anyway since the client ID is tied to using DigiD/eHerkenning flavours).